snapcraft/internal/remote_build/_worktree.py

```diff
@@ -14,100 +14,263 @@
+                        part_name, selector_source, selector=selector
+                    )
+                    if sinfo[selector] != selector_source:
+                        source_modified = True
+
+                part_yaml["source"].append(sinfo)
+        else:
+            part_yaml["source"] = self._pull_source(part_name, source)
+            if part_yaml["source"] != source:
+                source_modified = True
+
+        if source_modified:
+            # It's now an archive.
+            part_yaml["source-type"] = "tar"
+
+            # Strip source attributes no longer relevant.
+            for key in [
+                "source-branch",
+                "source-checksum",
+                "source-commit",
+                "source-depth",
+                "source-subdir",
+                "source-tag",
+            ]:
+                if key in part_yaml:
+                    part_yaml.pop(key)
+
+        return part_yaml
+
+    def _copy_assets(self):
+        assets_dir = self._project._get_snapcraft_assets_dir()
+        logger.debug("assets expected at: {}".format(assets_dir))
+
+        # Assets directory exists, but may not actually contain assets.
+        # The snapd project is an example of this.  The sanity checker
+        # will warn to the user when the project is built.
+        if os.path.exists(assets_dir):
+            logger.info(
+                "copying assets: {} -> {}".format(assets_dir, self._repo_snap_dir)
+            )
+            shutil.copytree(assets_dir, self._repo_snap_dir)
+        else:
+            os.makedirs(self._repo_snap_dir, exist_ok=True)
+
+        # Copy plugins if not already inside assets directory (and exists).
+        plugins_dir = self._project._get_local_plugins_dir()
+        if plugins_dir.startswith(assets_dir) or not os.path.exists(plugins_dir):
+            return
+
+        self._repo_plugins_dir = os.path.join(self._repo_snap_dir, "plugins")
+        logger.info(
+            "copying local plugins: {} -> {}".format(
+                plugins_dir, self._repo_plugins_dir
+            )
+        )
+        shutil.copytree(plugins_dir, self._repo_plugins_dir)
+
+    def _set_prepared_project_version(self):
+        """Set project version to work around two cases:
+        (1) `version: git` which uses git describe.
+        (2) `version-script: ...`
+
+        Both of these expect to run in project root (".").
+
+        TODO: We can probably better handle this by hinting
+        to snapcraft which parcat t to execute within, or to extract
+        a part to project root instead of its part directory.
+        """
+        if (
+            "version" in self._snapcraft_yaml
+            or "version-script" in self._snapcraft_yaml
+        ):
+            version = self._snapcraft_yaml.get("version", None)
+            version_script = self._snapcraft_yaml.get("version-script", None)
+            version = _version.get_version(version, version_script)
```

Comment:
version-script can be anything, it should just not be supported in this case

Reply:
Added error for remote build, opened PR to add DN #2639